File: utilities/geocoding.py
import urllib2
import json
import pymongo
import time

urlRoot = 'http://maps.googleapis.com/maps/api/geocode/json?address='

# Connection to Mongo DB
try:
    conn=pymongo.MongoClient()
    print "Connected successfully!!!"
except pymongo.errors.ConnectionFailure, e:
   print "Could not connect to MongoDB: %s" % e
conn

db = conn.finder
meetings = db.meetings
#print meetings.count()

cursor = meetings.find({ 'location' : None }).limit(500)

for doc in cursor:
    try:
        fullAddress = doc['address'] + '+' + doc['city'] + '+' + doc['state']
        cleanAddress = fullAddress.replace(' ', '+')
        url = urlRoot + cleanAddress
        response = urllib2.urlopen(url).read().decode('utf-8')
        json_obj = json.loads(response)
        if json_obj['status'] == 'OK':
            coor = json_obj['results'][0]['geometry']['location']
            print doc['_id']
            lng = coor['lng']
            lat = coor['lat']
            doc['location'] = [lng, lat]
            #db.test.update({ '_id': doc['_id'] }, { $set: { 'location[0]': lng, 'location[1]': lat } })
            db.meetings.save(doc)

        #sleep to throttle requests to api
        time.sleep(.3)

    except Exception:
        print 'Error: ' + str(doc['_id'])
        pass

File: utilities/geocoding.js
//var http = require('http');
//THIS DOESNT WORK.  USED PYTHON SCRIPT INSTEAD
var MongoClient = require('mongodb').MongoClient;
var request = require('request');

var urlRoot = 'http://maps.googleapis.com/maps/api/geocode/json?address=';

function executeRequest(db, url, doc) {
    request(url, function (error, response, body) {
        if (!error && response.statusCode == 200) {

            var data = JSON.parse(body);
            if (data.results && data.results.length > 0){
                var coordinates = data.results[0].geometry.location;
                var latitude = coordinates.lat;
                var longitude = coordinates.lng;

                db.collection('meetings').update({ _id: doc._id}, { location: [ longitude, latitude ] } );

                console.log('updated ' + doc._id);
            }
            //var latitude = data[0].geometry.location.lat;
            //var longitude = body.geometry.location.lng;
            //console.log(data);
        }
    })
}

var connString = 'mongodb://localhost:27017/finder';
MongoClient.connect(connString, function(err, db) {
    "use strict";
    if(err) throw err;
    console.log('geocoding script started');

    var cursor = db.collection('meetings').find({ location: { $exists: false } }).limit(10);
    cursor.forEach(function(doc){
        if (doc.address){
            var address = doc.address.replace(/ /g, '+') + '+' + doc.state;
            var url = urlRoot + address;
            //setTimeout(function(){ executeRequest(db, url, doc); }, 1000);
            request(url, function (error, response, body) {
                if (!error && response.statusCode == 200) {

                    var data = JSON.parse(body);
                    if (data.results && data.results.length > 0){
                        var coordinates = data.results[0].geometry.location;
                        var latitude = coordinates.lat;
                        var longitude = coordinates.lng;
                        console.log(latitude + ' ' + longitude);
                        db.collection('meetings').update({ _id: doc._id}, { location: [ longitude, latitude ] } );
                        console.log('updated ' + doc._id);
                    }
                }
            })
        }
});
});